Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py

```diff
@@ -4389,6 +4477,297 @@ def get_indicator_dbot_object(indicator):
+    limit = int(args.get('limit', 50))
+    page = arg_to_number(args.get('page'))
+    page = page - 1 if page else 0
+    page_size = int(args.get('page_size', 25))
+    page_size, limit = get_correct_page_size_and_limit(page_size, limit)
+    filter_req = create_filter([(machine_id, 'machineId'), (software_id, 'id'), (cve_id, 'cveId'), (fixing_kb_id, 'fixingKbId'),
+                                (product_name, 'productName'), (product_version, 'productVersion'), (severity, 'severity'),
+                                (product_vendor, 'productVendor')])
+    headers = ['id', 'cveId', 'machineId', 'productName', 'productVendor', 'productVersion', 'severity']
+    list_vulnerabilities_response = client.get_list_vulnerabilities_by_machine(filter_req, limit, page, page_size)
+    human_readable = tableToMarkdown(f'{INTEGRATION_NAME} vulnerabilities:',
+                                     list_vulnerabilities_response.get('value'), headers=headers, removeNull=True)
+    return CommandResults(
+        outputs_prefix='MicrosoftATP.MachineCVE',
+        outputs_key_field='id',
+        outputs=list_vulnerabilities_response.get('value'),
+        readable_output=human_readable,
+        raw_response=list_vulnerabilities_response)
+
+
+def create_filter_list_vulnerabilities(id_and_severity: str, name: str, description: str, published_on: str, cvss: str,
+                                       updated_on: str) -> str:
+    """ Create a string filter.
+        Args:
+            id_and_severity: str - Id and severity of the vulnerability.
+            name: str - Name of the vulnerability.
+            description: str - Description of the vulnerability.
+            published_on: str - Date when vulnerability was published..
+            cvss: str - CVSS v3 score.
+            updated_on: str - Date when vulnerability was updated.
+        Returns:
+            A string filter.
+    """
+    filter_query_list = []
+    if id_and_severity:
+        filter_query_list.append(id_and_severity)
+    if name:
+        filter_query_list.append(f"contains(name, '{name}')")
+    if description:
+        filter_query_list.append(f"contains(description, '{description}')")
+    if cvss:
+        filter_query_list.append(f"updatedOn ge {updated_on}")
+    if published_on:
+        filter_query_list.append(f"publishedOn ge {published_on}")
+
+    return create_filters_disjunctions(filter_query_list)
+
+
+def list_vulnerabilities_command(client: MsClient, args: dict) -> CommandResults:
+    """ Retrieves a list of all vulnerabilities.
+        Args:
+            client: MsClient.
+            args: dict - arguments from CortexSOAR.
+        Returns:
+            A CommandResults object.
+    """
+    id = argToList(args.get('id', ''))
+    severity = argToList(args.get('severity', ''))
+    name = args.get('name', '')
+    description = args.get('description', '')
+    published_on = args.get('published_on', '')
```

Comment:
Regarding all the timestamps args - create a helper function that gets the argument and transforms it to ISO format